broker/import_geo/routes.py

```diff
@@ -1,47 +1,88 @@
-# --- core imports
-from http import HTTPStatus
+import logging
 import re
 import tempfile
+from http import HTTPStatus
 
-# --- application imports
-from broker.common.util import response_json
-
-# --- third-party imports
 from flask import Blueprint, send_file, request
+from flask import current_app as app
 from flask_cors import cross_origin
 from geo_to_hca import geo_to_hca
+from ingest.api.ingestapi import IngestApi
+
+from ingest.importer.importer import XlsImporter
+from broker.common.util import response_json
+from broker.import_geo.exceptions import ImportGeoHttpError, InvalidGeoAccession, GenerateGeoWorkbookError, \
+    ImportProjectWorkbookError
 
 import_geo_bp = Blueprint(
     'import_geo', __name__, url_prefix='/'
 )
 
+LOGGER = logging.getLogger(__name__)
 
-def is_valid_geo_accession(geo_accession):
-    regex = re.compile("^GSE.*$")
-    return bool(regex.match(geo_accession))
 
-
-@import_geo_bp.route("/import-geo", methods=['POST'])
-@cross_origin(expose_headers=["Content-Disposition"])
+@import_geo_bp.route('/import-geo', methods=['POST'])
+@cross_origin(expose_headers=['Content-Disposition'])
 def get_spreadsheet_using_geo():
-    args = request.args
-    geo_accession = args.get('accession')
+    geo_accession = request.args.get('accession')
+
+    workbook = _generate_geo_workbook(geo_accession)
+    filename = f'hca_metadata_spreadsheet-{geo_accession}.xlsx'
+
+    return _send_file(filename, workbook)
+
+
+@import_geo_bp.route('/import-geo-project', methods=['POST'])
+@cross_origin(expose_headers=['Content-Disposition'])
+def import_project_using_geo():
+    geo_accession = request.args.get('accession')
+
+    workbook = _generate_geo_workbook(geo_accession)
 
-    if not is_valid_geo_accession(geo_accession):
-        return response_json(HTTPStatus.BAD_REQUEST, 'The given geo accession is invalid')
+    project_uuid = _import_project_from_workbook(workbook)
+
+    return response_json(HTTPStatus.OK, {'project_uuid': project_uuid})
+
+
+@import_geo_bp.errorhandler(ImportGeoHttpError)
+def handle_import_geo_http_error(e: ImportGeoHttpError):
+    return response_json(e.status_code, {'message': e.message})
+
+
+def _generate_geo_workbook(geo_accession: str):
+    if not _is_valid_geo_accession(geo_accession):
+        raise InvalidGeoAccession('The given geo accession is invalid.')
```

Comment:
I think the offending string would be useful here.

Reply:
Applied! thanks!